File: frontend/src/domain/entities/workflow-processing-job.ts
/**
 * Mirror of backend WorkflowProcessingJob presenter output. The backend
 * dispatcher endpoint (`POST /v1/workflows/{workflowId}/jobs`)
 * returns a subset of these fields (setId/temporalWorkflowId/status); the
 * full shape lands in the snapshot returned by the events endpoint.
 */

import type {
  WorkflowProcessingJobStatus,
  JobStep,
} from "@/src/domain/events/processing-job-event";

export interface WorkflowProcessingJobDocument {
  uuid: string;
  name: string;
  documentTypeId: string | null;
  documentIndex: number | null;
  pageRange: { from: number; to: number } | null;
  status: string;
  processingStatus: string | null;
}

export type WorkflowProcessingJobTrigger =
  | "USER"
  | "RETRY"
  | "ORPHAN_SWEEPER"
  | "SYSTEM";

export interface WorkflowProcessingJob {
  setId: string;
  temporalWorkflowId: string;
  workflowId: string;
  workflowCaseId?: string | null;
  fileId: string;
  fileName?: string | null;
  status: WorkflowProcessingJobStatus;
  currentStep?: JobStep | null;
  lastSeq: number;
  error?: string | null;
  resultSummary?: Record<string, unknown> | null;
  trigger?: WorkflowProcessingJobTrigger;
  createdById?: string | null;
  startedAt?: string | null;
  finishedAt?: string | null;
  durationMs?: number | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  documentCount?: number;
  documents?: WorkflowProcessingJobDocument[];
}

export interface WorkflowProcessingJobDispatchResponse {
  setId: string;
  temporalWorkflowId: string;
  status: WorkflowProcessingJobStatus;
}

export interface ProcessingJobFilters {
  search?: string;
  statuses?: string; // CSV: "COMPLETED,FAILED"
  workflowCaseId?: string;
  dateFrom?: string;
  dateTo?: string;
  cursor?: string;
  limit?: number;
}

export interface ProcessingJobPage {
  data: WorkflowProcessingJob[];
  pagination: {
    nextCursor: string | null;
    limit: number;
  };
}

File: backend/src/workflows/presentation/presenters/workflow_processing_job.py
from dataclasses import dataclass, field
from typing import Any

from src.common.application.helpers.uuids import optional_string
from src.common.application.helpers.datetimes import optional_datetime_string
from src.common.domain.interfaces.presenter import Presenter
from src.common.domain.models.processing.workflow_document import WorkflowDocument
from src.common.domain.models.workflow_processing_job import WorkflowProcessingJob


@dataclass
class WorkflowProcessingJobPresenter(Presenter[WorkflowProcessingJob]):
    instance: WorkflowProcessingJob
    file_name: str | None = None
    documents: list[WorkflowDocument] = field(default_factory=list)

    @property
    def to_dict(self) -> dict[str, Any]:
        return {
            "set_id": str(self.instance.uuid),
            "temporal_workflow_id": self.instance.temporal_workflow_id,
            "tenant_id": str(self.instance.tenant_id),
            "workflow_id": str(self.instance.workflow_id),
            "workflow_case_id": optional_string(self.instance.workflow_case_id),
            "file_id": str(self.instance.file_id),
            "file_name": self.file_name,
            "status": self.instance.status.value,
            "current_step": self.instance.current_step,
            "last_seq": self.instance.last_seq,
            "attempts": self.instance.attempts,
            "error": self.instance.error,
            "result_summary": self.instance.result_summary,
            "trigger": self.instance.trigger.value,
            "created_by_id": optional_string(self.instance.created_by_id),
            "started_at": optional_datetime_string(self.instance.started_at),
            "finished_at": optional_datetime_string(self.instance.finished_at),
            "duration_ms": self.instance.duration_ms,
            "document_count": len(self.documents),
            "documents": [
                {
                    "uuid": str(document.uuid),
                    "name": document.file_name,
                    "document_type_id": optional_string(document.document_type_id),
                    "document_index": document.document_index,
                    "page_range": document.page_range,
                    "status": str(document.status),
                    "processing_status": document.processing_status,
                }
                for document in self.documents
            ],
            "created_at": optional_datetime_string(self.instance.created_at),
            "updated_at": optional_datetime_string(self.instance.updated_at),
        }
